js/angular-edrna/SubConverter.js

```diff
@@ -0,0 +1,373 @@
+'use strict';
+
+(function() {
+	angular
+		.module('edrnaApp')
+		.provider('SubConverter', SubConverterProvider)
+		.config(SubConverterCfg);
+
+	SubConverterCfg.$inject = ['SubConverterProvider'];
+
+	function SubConverterProvider() {
+		var srcConverters = {};
+		var dstConverters = {};
+
+		this.addSrcConverter = addSrcConverter;
+		this.addDstConverter = addDstConverter;
+		this.$get = ['$q', function ($q) {
+			return new SubConverter(srcConverters, dstConverters, $q);
+		}];
+
+		function addSrcConverter(mime, converter) {
+			if(typeof mime !== 'string' || typeof converter !== 'function') {
+				console.error('Converter plug-in is not a function!');
+				return false;
+			}
+
+			srcConverters[mime] = converter;
+		}
+
+		function addDstConverter(mime, converter) {
+			if(typeof mime !== 'string' || typeof converter !== 'function') {
+				console.error('Converter plug-in is not a function!');
+				return false;
+			}
+
+			dstConverters[mime] = converter;
+		}
+	}
+
+	function SubConverter(srcConverters, dstConverters, $q) {
+		var supportedSrc = Object.keys(srcConverters);
+		var supportedDst = Object.keys(dstConverters);
+
+		return {
+			newConverter: newConverter,
+			isSupportedSrc: isSupportedSrc,
+			isSupportedDst: isSupportedDst
+		};
+
+		function newConverter() {
+			return new converter();
+		}
+
+		function isSupportedSrc(mime) {
+			return supportedSrc.indexOf(mime) !== -1;
+		}
+
+		function isSupportedDst(mime) {
+			return supportedDst.indexOf(mime) !== -1;
+		}
+
+		function converter() {
```

Comment:
That function is a constructor function and it's name should start from upper cased _C_ as for rest of constructors there